src/taco.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  Ciphertext,
+  DkgPublicKey,
+  ferveoEncrypt,
+} from '@nucypher/nucypher-core';
+import { ethers } from 'ethers';
+
+import { ThresholdDecrypter } from './characters/cbd-recipient';
+import { Condition, ConditionExpression } from './conditions';
+import { DkgClient } from './dkg';
+import { getPorterUri } from './porter';
+import { toBytes } from './utils';
+
+export interface TacoMessageKit {
```

Comment:
Was this a stand-in until `nucypher-core` 0.12.0 was released? Any reason why `nucypher-core` `ThresholdMessageKit` cannot be used?

Reply:
Yes, but it seems to be that we still need some extra information not covered by `ThresholdMessageKit`: We need to know the `ritualId` and `threshold` to construct `DecryptionRequests`.

Reply:
This is covered via latest changes in the Coordinator contract.

There is/will be (once deployed) the ability to get the ritual id from the public key via the Coordinator contract - see https://github.com/nucypher/nucypher-contracts/pull/107/files#diff-044dc40960596bb6ed1e69a1a50bfff628a2ca07775e737eec923b13e857fb43R366 (https://github.com/nucypher/nucypher-contracts/pull/107).

`threshold` is also stored in the Ritual stuct in the Coordinator as part of that change as well; once you know the ritual id, you can get the threshold https://github.com/nucypher/nucypher-contracts/pull/107/files#diff-044dc40960596bb6ed1e69a1a50bfff628a2ca07775e737eec923b13e857fb43R67.